base/server/python/pki/server/__init__.py

```diff
@@ -247,6 +247,52 @@ def restart(self):
         logger.debug('Command: %s', ' '.join(cmd))
         subprocess.check_call(cmd)
 
+    def run(self, as_current_user=False):
+
+        java_home = self.config['JAVA_HOME']
```

Comment:
I don't think this is quite correct. I get:

```
# pki-server run tomcat@pki
ERROR: 'JAVA_HOME'
```

So then I try:

```
# JAVA_HOME="/usr/lib/jvm/jre" pki-server run tomcat@pki
```

Perhaps I'm just not sure what this `config` object points to? `pki-server --help` and `pki-server run --help` don't seem to tell me.

Reply:
The `self.config` is actually loaded from `/etc/sysconfig/<instance>` by [this code](https://github.com/dogtagpki/pki/blob/1035cdc3e0ee63ab87a5faa4a4e9d43a3e3a421d/base/server/python/pki/server/__init__.py#L516). So to change JAVA_HOME or any other variables, it has to be done in that file. That way the changes will be consistent regardless whether the server is run via command line with this script or as a systemd service.

Note, currently there is another copy of that file at `/etc/pki/<instance>/tomcat.conf` and they need to be identical. Ideally there should be just one file.

Reply:
I followed the [documentation](https://github.com/dogtagpki/pki/blob/master/docs/installation/Installing_Basic_PKI_Server.md#creating-tomcat-instance) in a container (Fedora 30).

My `/etc/sysconfig/tomcat@pki` is:
```
# Service-specific configuration file for tomcat. This will be sourced by
# systemd for the default service (tomcat.service)
# If you want to customize named instance, make a similar file
# and name it tomcat@instancename.

# You will not need to set this, usually. For default service it equals
# CATALINA_HOME. For named service, it equals ${TOMCATS_BASE}${NAME}
#CATALINA_BASE="/usr/share/tomcat"

# Please take a look at /etc/tomcat/tomcat.conf to have an idea what you
# can override.
CATALINA_BASE="/var/lib/tomcats/pki"
```

I've added a [`Dockerfile`](https://github.com/cipherboy/containers/blob/master/devel/pki_pull_210) to my [`containers`](https://github.com/cipherboy/containers) repo.

Reply:
Sure, looks better. Are `JAVA_HOME` and especially `JAVA_OPTS` strictly required, or are they optional? The latter seems optional; perhaps `self.config.get('JAVA_OPTS', "")` might be better. 

At any rate, we can clean up error handling/usability of these configs in a different patch as it'd likely affect more than just `pki-server run`.

Reply:
The systemd unit file will always include the default /etc/tomcat/tomcat.conf which defines JAVA_HOME and JAVA_OPTS, so the variables are always set. We could certainly add error handling in case the admin customizes the default tomcat.conf, but that's probably unlikely to happen.

Reply:
Okie dokie, sounds fine to me. You can merge if you want.